apps/web/modules/auth/signup/actions.ts

```diff
@@ -93,47 +93,47 @@ export const createUserAction = actionClient.schema(ZCreateUserAction).action(as
     await deleteInvite(invite.id);
   }
-  // Handle organization assignment
-  else {
-    let organizationId: string | undefined;
-    let role: TOrganizationRole = "owner";
-
-    if (parsedInput.defaultOrganizationId) {
-      // Use existing or create organization with specific ID
-      let organization = await getOrganization(parsedInput.defaultOrganizationId);
-      if (!organization) {
-        organization = await createOrganization({
-          id: parsedInput.defaultOrganizationId,
-          name: `${user.name}'s Organization`,
-        });
-      } else {
-        role = parsedInput.defaultOrganizationRole || "owner";
-      }
-      organizationId = organization.id;
-    } else {
-      const isMultiOrgEnabled = await getIsMultiOrgEnabled();
-      if (isMultiOrgEnabled) {
-        // Create new organization
-        const organization = await createOrganization({ name: `${user.name}'s Organization` });
-        organizationId = organization.id;
-      }
-    }
 
-    if (organizationId) {
-      await createMembership(organizationId, user.id, { role, accepted: true });
-      await updateUser(user.id, {
-        notificationSettings: {
-          ...user.notificationSettings,
-          alert: { ...user.notificationSettings?.alert },
-          weeklySummary: { ...user.notificationSettings?.weeklySummary },
-          unsubscribedOrganizationIds: Array.from(
-            new Set([...(user.notificationSettings?.unsubscribedOrganizationIds || []), organizationId])
-          ),
-        },
+  let organizationId: string | undefined;
+  let role: TOrganizationRole = "owner";
+
+  if (parsedInput.defaultOrganizationId) {
+    // Use existing or create organization with specific ID
+    let organization = await getOrganization(parsedInput.defaultOrganizationId);
+    if (!organization) {
+      organization = await createOrganization({
+        id: parsedInput.defaultOrganizationId,
+        name: `${user.name}'s Organization`,
       });
+    } else {
+      role = parsedInput.defaultOrganizationRole || "owner";
+    }
+    organizationId = organization.id;
+  } else {
+    const isMultiOrgEnabled = await getIsMultiOrgEnabled();
+    if (isMultiOrgEnabled) {
+      // Create new organization
+      const organization = await createOrganization({ name: `${user.name}'s Organization` });
+      organizationId = organization.id;
     }
   }
 
+  if (organizationId) {
+    await createMembership(organizationId, user.id, { role, accepted: true });
+    await updateUser(user.id, {
+      notificationSettings: {
+        ...user.notificationSettings,
+        alert: { ...user.notificationSettings?.alert },
+        weeklySummary: { ...user.notificationSettings?.weeklySummary },
+        unsubscribedOrganizationIds: Array.from(
+          new Set([...(user.notificationSettings?.unsubscribedOrganizationIds || []), organizationId])
+        ),
+      },
+    });
+
+    await createDefaultTeamMembership(user.id);
```

Comment:
I think we should first check if a `DEFAULT_TEAM_ID` is set in the environment variables before calling this service instead of having the check inside the service 🙏